### PR TITLE
fix getSchemePermissions

### PR DESCRIPTION
**Pull request by dkent600**
Fixes bug in `getSchemePermissions` and adds a couple other minor enhancements.

### CODE REVIEW

File: lib/contractWrapperBase.ts [hunk 1]
```diff
@@ -168,7 +168,7 @@ export abstract class ContractWrapperBase {
    * @param avatarAddress
    */
   protected async _getSchemePermissions(avatarAddress: Address): Promise<SchemePermissions> {
-    const permissions = (await this.getController(avatarAddress))
+    const permissions = await (await this.getController(avatarAddress))
```
Comment (dev-matan-tsuberi) [inline lib/contractWrapperBase.ts]: why the double `await`?

Comment (dkent600) [reply lib/contractWrapperBase.ts]: there are two async functions being called here.  the bug is that I wasn't awaiting the second one

Comment (dev-matan-tsuberi) [reply lib/contractWrapperBase.ts]: But when you return a promise from a `then` callback, it flattens it into a single promise. Ther'e can't be a promise of a promise (`Promise<Promise<T>>`).

Comment (dkent600) [reply lib/contractWrapperBase.ts]: The parens ensure that it is not a promise of a promise.  See also the test I added that confirms that the function works now.

Comment (dev-matan-tsuberi) [reply lib/contractWrapperBase.ts]: single `await` should work just as well as double `await`.

Comment (dkent600) [reply lib/contractWrapperBase.ts]: It doesn't because you end up getting a promise from the second async call instead of the actual value being promised.

**Pull request merged**